### PR TITLE
refactor(JP-KY): remove arrow usage

### PR DESCRIPTION
**Pull request by shuuji3**
## Issue

#6135

## Description

- This PR removes `arrow` from `parsers/JP-KY.py`. (fa9f0b0f0731e12ab353fa16c863facb4ed7f69a)
- Also, I did some refactoring on the code to improve code readability. (72ca498896bbe114002a6699217267109a2a61be) (I run the test before/after the refactoring so the behavior should not be changed.)

### Preview

N/A

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/JP-KY.py [hunk 1]
```diff
@@ -40,81 +41,85 @@ def fetch_production(
...
-    minutes = int(re.findall(r"(?<=時)[\d]+(?=分)", ts)[0])
+
     # get date
-    ds = soup.find("div", class_="puChangeGraph")
-    date = re.findall(r"(?<=chart/chart)[\d]+(?=.gif)", str(ds))[0]
+    date_div = soup.find("div", class_="puChangeGraph")
+    date_str = re.findall(r"(?<=chart/chart)[\d]+(?=.gif)", str(date_div))[0]
+    year_str = date_str[:4]
+    month_str = date_str[4:6]
+    day_str = date_str[6:]
```
Comment (VIKTORVAV99) [inline parsers/JP-KY.py:57]: Why do we need to split them up into all these strings?
Cant we just parse it as a date and then use that?

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM!

Just one tiny comment I'd like a response to, but I think we can merge this regardless.


PS: Thanks for removing all the abbreviations.

File: parsers/JP-KY.py (hunk 1 above)
Comment (shuuji3) [reply parsers/JP-KY.py:57]: Ah, that's correct. I was influenced by the original code a bit.

I'll adjust that date parsing soon.

Comment (VIKTORVAV99) [reply parsers/JP-KY.py]: Yeah I'm aware and I want to update but some of the libs we use in the backend + the version of pandas we use (I think) don't support Python 3.11 yet and we want to keep them in sync.

It's really annoying though 😅

Comment (shuuji3) [reply parsers/JP-KY.py]: That makes sense😅 I'm OK with 3.10 for now. Hopefully these dependencies will gradually adopt to newer versions.

**Pull request merged**